src/array/primitive/mutable.rs

```diff
@@ -137,6 +137,25 @@ impl<T: NativeType> MutablePrimitiveArray<T> {
         }
     }
 
+    /// Pop a value from the array.
+    /// Note if the values is empty, this method will return None.
+    pub fn pop(&mut self) -> Option<T> {
+        if self.values.is_empty() {
```

Comment:
We could match on `self.values.pop()` instead, which does itself perform the same check.

```rust
let value = self.values.pop()?;
...
```

Reply:
When we get `None` from `self.values.pop()`, there are two conditions:
1. the `array` is empty.
2. the last value of the `array` is None.

So when we match on `self.values.pop()?` first, it will skip `validity.pop()` in the `2nd` condition.

Reply:
I'm not saying we should skip matching on the validity, but it seems to me we can replace L143-146 by `let value = self values.pop()?`.

Reply:
If the last value of array is `None`, that means we need to handle `validity bitmap` next, 
but `let value = self.values.pop()?;` will directly return without handle the `validity bitmap` in this condition.

Reply:
`self.values` is of type `Vec<T>`, where `T` is a primitive type, so by definition it will not return a `None` from this `Vec<T>` itself (it will return `None` on a empty `Vec`).

Reply:
@ygf11, @Dandandan argument holds because `self.values.len() == self.validity.len()` when the validity exists (it is an invariant of the array)

Reply:
I'm sorry, I recognized `Vec<T>` as `Vec<Option<T>>` here.

Reply:
Addressed.